framework/src/dgkernels/DGKernel.C

```diff
@@ -65,7 +65,7 @@ DGKernel::DGKernel(const InputParameters & parameters)
     FunctionInterface(this),
     UserObjectInterface(this),
     NeighborCoupleableMooseVariableDependencyIntermediateInterface(this, false, false),
-    TwoMaterialPropertyInterface(this),
+    TwoMaterialPropertyInterface(this, boundaryRestricted() ? std::set<SubdomainID>() : blockIDs(), boundaryIDs()),
```

Comment:
I checked the failing test in Zapdos. Nothing wrong there. For interface kernels, the coverage check is much more complicated. We need to call `markMatPropRequested` differently in `getMaterialProperty` and `getNeighborMaterialProperty` based on the subdomains attaching the internal boundary. The material property could be provided by block or boundary materials on the master side, which makes things more complicated. @permcody has an idea in his mind that we the coverage check in front by looping though elements and sides and deprecated the current checking method.

Thus, I have `boundaryRestricted() ? std::set<SubdomainID>() : blockIDs(), boundaryIDs()` here as a heck to enable the check for `DGKernel` and `InternalSideUserObject` but skip the check for `InterfaceKernel`. I think this is worthwhile to be pushed into framework, but need to have others' opinion.